crates/requirements-txt/src/requirement.rs

```diff
@@ -1,74 +1,85 @@
-use pep508_rs::{
-    MarkerEnvironment, MarkerTree, Pep508Error, Pep508ErrorSource, Requirement, UnnamedRequirement,
-    VersionOrUrl, VersionOrUrlRef,
-};
-use serde::{Deserialize, Serialize};
 use std::fmt::{Display, Formatter};
 use std::path::Path;
-use std::str::FromStr;
+
+use thiserror::Error;
+
+use distribution_types::{ParsedUrl, ParsedUrlError, RequirementSource};
+use pep508_rs::{
+    MarkerEnvironment, MarkerTree, Pep508Error, Pep508ErrorSource, UnnamedRequirement,
+};
 use uv_normalize::ExtraName;
 
+use crate::Requirement;
+
 /// A requirement specifier in a `requirements.txt` file.
-#[derive(Hash, Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
+#[derive(Hash, Debug, Clone, Eq, PartialEq)]
 pub enum RequirementsTxtRequirement {
-    /// A PEP 508-compliant dependency specifier.
-    Pep508(Requirement),
+    /// The uv-specific superset over PEP 508 requirements specifier incorporating
+    /// `tool.uv.sources`.
+    Uv(Requirement),
     /// A PEP 508-like, direct URL dependency specifier.
     Unnamed(UnnamedRequirement),
 }
 
 impl Display for RequirementsTxtRequirement {
     fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
         match self {
-            Self::Pep508(requirement) => write!(f, "{requirement}"),
+            Self::Uv(requirement) => write!(f, "{requirement}"),
             Self::Unnamed(requirement) => write!(f, "{requirement}"),
         }
     }
 }
 
 impl RequirementsTxtRequirement {
+    /// For error messages.
+    pub fn name_or_url(&self) -> String {
+        match self {
+            RequirementsTxtRequirement::Uv(requirement) => requirement.name.to_string(),
+            RequirementsTxtRequirement::Unnamed(unnamed) => unnamed.to_string(),
+        }
+    }
```

Comment:
Why is this not implementing display?

Reply:
Which part? There's a `impl Display for RequirementsTxtRequirement`